projects/step-core/src/lib/modules/table/components/custom-columns/custom-columns.component.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  Component,
+  EventEmitter,
+  Input,
+  OnChanges,
+  Output,
+  QueryList,
+  SimpleChanges,
+  TrackByFunction,
+  ViewChildren,
+} from '@angular/core';
+import { AugmentedScreenService } from '../../../../client/augmented/step-augmented-client.module';
+import { Input as ColInput } from '../../../../client/generated';
+import { map } from 'rxjs';
+import { MatColumnDef } from '@angular/material/table';
+import { SearchColDirective } from '../../directives/search-col.directive';
+
+@Component({
+  selector: 'step-custom-columns',
+  templateUrl: './custom-columns.component.html',
+  styleUrls: ['./custom-columns.component.scss'],
+})
+export class CustomColumnsComponent implements OnChanges {
+  @Input() screen!: string;
+  @Input() excludeFields?: string[];
+  @Input() isSearchDisabled?: boolean;
+
+  @Output() columnsReady = new EventEmitter<unknown>();
+
+  columns: ColInput[] = [];
+
+  readonly trackByCol: TrackByFunction<ColInput> = (index, item) => item.id;
+
+  @ViewChildren(MatColumnDef) colDef?: QueryList<MatColumnDef>;
+  @ViewChildren(SearchColDirective) searchColDef?: QueryList<SearchColDirective>;
+
+  constructor(private _screens: AugmentedScreenService) {}
```

Comment:
the name `_screens` lets me think of an array of objects or primitives. in this case `screen` even is a primitive. For the injected service, a name like `_screensApiService` would make the code more readable imo